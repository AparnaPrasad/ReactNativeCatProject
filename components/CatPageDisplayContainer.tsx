import React from 'react';
import { View, Image } from 'react-native';

interface IOwnProps {
    imageWidth: number,
    header?: React.ReactNode,
    content?: React.ReactNode
}

const CatPageDisplayContainer = ({ imageWidth, header, content }: IOwnProps) => {
    return (<View style={{ flex: 1}}>
        <Image
            style={{ width: '100%', height: '50%' }}
            source={{ uri: `https://cataas.com/cat?width=${imageWidth}&height=149` }} />
        <View>
            {header}
        </View>
        <View>
            {content}
        </View>
        
    </View>)
}

export default CatPageDisplayContainer;