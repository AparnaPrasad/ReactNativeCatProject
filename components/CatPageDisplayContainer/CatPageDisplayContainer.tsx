import React from 'react';
import { View, Image, StyleSheet, ScrollView } from 'react-native';

interface IOwnProps {
    imageWidth: number,
    header?: React.ReactNode,
    content?: React.ReactNode,
    imageUrl: string
}

const CatPageDisplayContainer = ({ imageWidth, header, content, imageUrl }: IOwnProps) => {
    return (<View style={styles.imageAndContentContainer}>
        <Image
            testID='catImage'
            style={styles.catImageContainer}
            source={{ uri: `${imageUrl}&width=${imageWidth}` }} />
        <View style={styles.contentContainer}>
            <View style={styles.headerPadding}>
                {header}
            </View>
            <ScrollView style={styles.scrollStyle}>
                {content}
            </ScrollView>
        </View>
    </View>)
}

export default CatPageDisplayContainer;

const styles = StyleSheet.create({
    imageAndContentContainer: {
        flex: 1
    },
    scrollStyle: {
        flex: 1
    },
    catImageContainer: {
        width: '100%',
        height: '50%'
    },
    contentContainer: {
        padding: 20,
        height: '50%'
    },
    headerPadding: {
        paddingVertical: 20
    }
})