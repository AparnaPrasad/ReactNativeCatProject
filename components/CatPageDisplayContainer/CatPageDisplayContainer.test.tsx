import { shallow } from "enzyme";
//import Image from 'react-native';
import React from 'react';
import CatPageDisplayContainer from './CatPageDisplayContainer';

describe('CatPageDisplayContainer should render correctly', () => {

    it('should render image with given url and given wodth', () => {
        const imageUrl = 'testurl';
        const width = 50;
        const expectedUrl = `${imageUrl}&width=${width}`
        const wrapper = shallow(<CatPageDisplayContainer imageUrl={imageUrl} content='test content' header='test header' imageWidth={width} />)
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find("Image").prop("source")).toEqual({ uri: expectedUrl});
    })
});