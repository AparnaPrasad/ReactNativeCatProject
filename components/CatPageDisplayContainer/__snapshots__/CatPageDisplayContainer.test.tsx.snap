// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CatPageDisplayContainer should render correctly should render image with given url and given wodth 1`] = `ShallowWrapper {}`;
